acceptance/helpers.rb

```diff
@@ -0,0 +1,130 @@
+require 'beaker-puppet'
+
+module Beaker
+  module DSL
+   module Roles
+      # @return [Array<Host>] Hosts with the agent role, except for the master. May be empty.
+      def agents_only
+        hosts_as(:agent).select do |host|
+          !host['roles'].include?('master')
+        end.to_a
+      end
+    end
+  end
+end
+
+# Gather arguments for `install_puppet_agent_on` based on the environment, sanity check, and format them.
+# @return [Hash] An options hash to pass to BeakerPuppet's `install_puppet_agent_on` method
+def agent_install_options
+  agent_version = ENV['AGENT_VERSION'] || ENV['PUPPET_CLIENT_VERSION'] # This is the legacy name from puppet 3 / module 1.x
+
+  if agent_version
+    unless dev_builds_accessible?
+      # The user requested a specific build, but they can't download from internal sources
+      env_var_name = ENV['AGENT_VERSION'] ? 'AGENT_VERSION' : 'PUPPET_CLIENT_VERSION'
+      fail_test(<<-WHY
+  You requested a specific build of puppet-agent, but you don't have access to
+  Puppet's internal build servers. You can either:
+
+  - Unset the #{env_var_name} environment variable to accept the latest Puppet 4
+    agent release (this is the defualt), or
+  - Set the $PUPPET_COLLECTION environment variable to 'puppet5' or 'puppet6' to
+    use the latest releases from those streams.
+
+  WHY
+      )
+    end
+
+    return { puppet_agent_version: agent_version }
+  end
+
+  { puppet_collection: (ENV['PUPPET_COLLECTION'] || 'pc1').downcase }
+end
+
+# Installs the puppet_agent module on the target host, and then uses the PMT to install puppet_agent's dependencies
+def install_modules_on(host)
+  install_dev_puppet_module_on(host, {
+    source: File.join(File.dirname(__FILE__), '..', ),
+    module_name: 'puppet_agent'
+  })
+
+  on(host, puppet('module', 'install', 'puppetlabs-stdlib',     '--version', '4.16.0'), { acceptable_exit_codes: [0] })
+  on(host, puppet('module', 'install', 'puppetlabs-inifile',    '--version', '2.1.0'),  { acceptable_exit_codes: [0] })
+  on(host, puppet('module', 'install', 'puppetlabs-apt',        '--version', '4.4.0'),  { acceptable_exit_codes: [0] })
+  on(host, puppet('module', 'install', 'puppetlabs-transition', '--version', '0.1.1'),  { acceptable_exit_codes: [0] })
```

Comment:
I'd rather read these module versions from ../metadata.json, but I figured I'd save that for later, since we're updating them soon anyway.

Reply:
Maybe add a TODO then?